src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1726,4 +1726,22 @@ impl Cage {
             syscall_error(Errno::EBADF, "getdents", "Invalid file descriptor")
         }
     }
+
+    //------------------------------------GETCWD SYSCALL------------------------------------
+    
+    pub fn getcwd_syscall(&self, buf: *mut u8, bufsize: u32) -> i32 {
+        let cwd: String = format!("{}{}", self.cwd.read().unwrap().to_string_lossy(), "\0");
+        let bytes: Vec<u8> = cwd.as_bytes().to_vec();
```

Comment:
I don't think this is a very good implementation. We'd prefer to_string to to_string_lossy. Then you'd just have an &str which you can use as_bytes on directly. without needing to format anything. To add the terminating null byte, just push a 0 onto the stack.

Reply:
Updated.